onnxruntime/core/common/threadpool.cc

```diff
@@ -57,6 +57,29 @@ void ThreadPool::ParallelFor(int32_t total, std::function<void(int32_t)> fn) {
   barrier.Wait();
 }
 
+void ThreadPool::BatchParallelFor(int32_t total, std::function<void(int32_t)> fn, int32_t batch_size) {
+  if (total <= 0)
+    return;
+
+  if (total == 1 || batch_size <= 1) {
+    fn(0);
+    return;
```

Comment:
Unexpected that I could have a total of say 5 and a batch size of 1 and it would only call fn once. Shouldn't fn always be called 'total' times regardless of batch size?

Reply:
it is possible that if a task is small enough, it's faster to do the 5 tasks in one thread in sequence, than to dispatch that to 5 threads.

Reply:
if the caller thinks it's a good idea to do that in one batch, there is no reason not to do so.